app/controllers/organization_wizzard_controller.rb

```diff
@@ -0,0 +1,77 @@
+class OrganizationWizzardController < ApplicationController
+  layout 'settings'
+  before_action :set_organization
+  before_action :set_organization_wizzard
+  before_action :redirect_if_complite_second_step, only: [ :new_account,
+    :create_default_accounts, :create_accounts ]
+  before_action :redirect_if_complite_third_step, only: [ :new_category, 
+    :create_default_categories, :create_categories ]
+
+  def new_account
+    @organization.bank_accounts.build
+  end
+
+  def new_category
+    @organization.categories.build
+  end
+
+  def create_default_accounts
+    BankAccount.create_defaults(@organization)
+    redirect_to new_category_organization_path
+  end
+
+  def create_default_categories
+    Category.create_defaults(@organization)
+    session[:new_organization] = true
+    redirect_to root_path
+  end
+
+  def create_accounts
+    if @organization.update(organization_params)
+      redirect_to :new_category, notice: "Bank accounts was created successfully."
+    else
+      flash[:open_form] = true
+      render :new_account
+    end
+  end
+
+  def create_categories
+    if @organization.update(organization_params)
+      flash[:new_organization] = true
+      redirect_to root_path, notice: "Categories was created successfully."
+    else
+      flash[:open_form] = true
+      render :new_category
+    end
+  end
+
+  private
+
+  def set_organization
+    @organization = current_organization
+  end
+
+  def set_organization_wizzard
+    @organization_wizzard = OrganizationWizzard.new(@organization)
+  end
+
+  def organization_params
+    params.require(:organization).permit(bank_accounts_attributes: [:id,
```

Comment:
@dissident Make separate `..._params` methods for bank accounts and categories. Because you have separate actions
```
def bank_accounts_params
   params.require(:organization).permit(bank_accounts_attributes: [...])
end

def categories_params
   params.require(:organization).permit(categories_attributes: [...])
end